virtualhome/grails-app/views/templates/_passwordinput.gsp

```diff
@@ -77,16 +77,16 @@
     }
   }, "");
 
-  jQuery.validator.addMethod("noSpace", function(value, element) { 
-    return value.indexOf(" ") < 0 && value != ""; 
-  }, "Spaces are not permitted.");
+  jQuery.validator.addMethod("matchesPattern", function(value, element) {
+    return value.match(/^[_a-zA-Z][-_.a-zA-Z0-9]*$/) && value != "";
+  }, "Not a permitted username.");
```

Comment:
"Not a permitted username" doesn't read quite right to me. Maybe something like

```suggestion
  }, "Username is not valid.");
```

Reply:
Thanks - that reads much better!